src/test/java/io/supertokens/test/IpAllowDenyRegexTest.java

```diff
@@ -353,4 +360,89 @@ public void CheckNoLoggingForNotAllowedAPIRoutes() throws Exception {
+                    new JsonObject()
+            ), false);
+
+            // this should pass
+            HttpRequest.sendGETRequest(process.getProcess(), "", "http://localhost:3567/hello", null, 1000, 1000,
+                    null);
+            HttpRequest.sendGETRequest(process.getProcess(), "", "http://localhost:3567/t2/hello", null, 1000, 1000,
+                    null);
+
+            try {
+                HttpRequest.sendGETRequest(process.getProcess(), "", "http://localhost:3567/t1/hello", null, 1000, 1000,
+                        null);
+                fail();
+            } catch (HttpResponseException e) {
+                assertEquals(e.statusCode, 403);
+            }
+
+            process.kill();
+            assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+        }
+
+        { // test deny works
+            Utils.reset();
+            TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+            FeatureFlagTestContent.getInstance(process.getProcess())
+                    .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{EE_FEATURES.MULTI_TENANCY});
+            process.startProcess();
+            assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+            JsonObject coreConfig = new JsonObject();
+            coreConfig.addProperty("ip_deny_regex", "127\\.\\d+\\.\\d+\\.\\d+|::1|0:0:0:0:0:0:0:1");
+
+            Multitenancy.addNewOrUpdateAppOrTenant(process.getProcess(), new TenantConfig(
+                    new TenantIdentifier(null, null, "t1"),
+                    new EmailPasswordConfig(true), new ThirdPartyConfig(true, null), new PasswordlessConfig(true),
+                    coreConfig
+            ), false);
+            Multitenancy.addNewOrUpdateAppOrTenant(process.getProcess(), new TenantConfig(
+                    new TenantIdentifier(null, null, "t2"),
+                    new EmailPasswordConfig(true), new ThirdPartyConfig(true, null), new PasswordlessConfig(true),
+                    new JsonObject()
+            ), false);
+
+            // this should pass
+            HttpRequest.sendGETRequest(process.getProcess(), "", "http://localhost:3567/hello", null, 1000, 1000,
+                    null);
+            HttpRequest.sendGETRequest(process.getProcess(), "", "http://localhost:3567/t2/hello", null, 1000, 1000,
+                    null);
+
+            try {
+                HttpRequest.sendGETRequest(process.getProcess(), "", "http://localhost:3567/t1/hello", null, 1000, 1000,
+                        null);
+                fail();
+            } catch (HttpResponseException e) {
+                assertEquals(e.statusCode, 403);
+            }
+
+            process.kill();
+            assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+        }
```

Comment:
add a test where if ip allow is for an app, it's applied to all tenants for that app and to no other app

Reply:
Done